src/native/containers/dn-simdhash-specialization.h

```diff
@@ -471,18 +464,19 @@ DN_SIMDHASH_TRY_ADD_WITH_HASH (DN_SIMDHASH_T_PTR hash, DN_SIMDHASH_KEY_T key, ui
 	switch (ok) {
 		case DN_SIMDHASH_INSERT_OK_ADDED_NEW:
 			hash->count++;
-			return 1;
+			return DN_SIMDHASH_ADD_INSERTED;
 		case DN_SIMDHASH_INSERT_OK_OVERWROTE_EXISTING:
 			// This shouldn't happen
 			dn_simdhash_assert(!"Overwrote an existing item while adding");
-			return 1;
+			return DN_SIMDHASH_ADD_OVERWROTE;
```

Comment:
Isn't this code unreachable ? Should it be `DN_SIMDHASH_ADD_FAILED` for clarity ?

Reply:
dn_simdhash_assert is configurable, so it might not actually terminate execution. good point that in this case we should signal add failure instead of an overwrite.